htxs.hxx

```diff
@@ -0,0 +1,115 @@
+ * @param htxs_pth Name of the column with pt(H) from the htxs module.
+ * @param htxs_njets Name of the column with the number of jets from the htxs
+ * module.
+ * @returns a dataframe with the weight column included
+ */
+auto ggHNLLOWeights(auto &df, const std::string &weight_name,
+                    const std::string &rootfilename,
+                    const std::string &generator, const std::string &htxs_pth,
+                    const std::string &htxs_njets) {
+    TGraphErrors *WeightsGraphs[4];
+    TFile rootFile(rootfilename.c_str(), "READ");
+    if (generator == "powheg") {
+        WeightsGraphs[0] =
+            (TGraphErrors *)rootFile.Get("gr_NNLOPSratio_pt_powheg_0jet");
+        WeightsGraphs[1] =
+            (TGraphErrors *)rootFile.Get("gr_NNLOPSratio_pt_powheg_1jet");
+        WeightsGraphs[2] =
+            (TGraphErrors *)rootFile.Get("gr_NNLOPSratio_pt_powheg_2jet");
+        WeightsGraphs[3] =
+            (TGraphErrors *)rootFile.Get("gr_NNLOPSratio_pt_powheg_3jet");
+    } else if (generator == "amcatnlo") {
+        WeightsGraphs[0] =
+            (TGraphErrors *)rootFile.Get("gr_NNLOPSratio_pt_mcatnlo_0jet");
+        WeightsGraphs[1] =
+            (TGraphErrors *)rootFile.Get("gr_NNLOPSratio_pt_mcatnlo_1jet");
+        WeightsGraphs[2] =
+            (TGraphErrors *)rootFile.Get("gr_NNLOPSratio_pt_mcatnlo_2jet");
+        WeightsGraphs[3] =
+            (TGraphErrors *)rootFile.Get("gr_NNLOPSratio_pt_mcatnlo_3jet");
+    } else
+        Logger::get("ggHNLLOWeights")
+            ->critical("WARNING: Invalid ggH generator configured. "
+                       "ggHNNLOWeights cannot be determined!");
+    rootFile.Close();
+    const Float_t cutoff[4] = {125.0, 625.0, 800.0, 925.0};
+    auto readout_lambda = [WeightsGraphs, cutoff](const Float_t &htxs_pth,
+                                                  const UChar_t &htxs_njets) {
+        int njets = std::min(3, int(htxs_njets));
+        return WeightsGraphs[njets]->Eval(std::min(htxs_pth, cutoff[njets]));
+    };
+    auto df1 = df.Define(weight_name, readout_lambda, {htxs_pth, htxs_njets});
+    return df1;
+}
+
+/**
+ * @brief Function to derive the WG1 ggH uncertainties
+ *
+ * @param weight_names Names of the derived weight in the dataframe in the order
+ * given by the WG1 macro.
+ * @param htxs_flag Name of the column with the htxs stage1 (NOT 1.1, 1.2 or
+ * later!) flag.
+ * @param htxs_pth Name of the column with pt(H) from the htxs module.
+ * @param htxs_njets Name of the column with the number of jets from the htxs
+ * module.
+ * @returns a dataframe with the weight column included
+ */
+auto ggH_WG1_uncertainties(auto &df,
+                           const std::vector<std::string> &weight_names,
+                           const std::string &htxs_flag,
+                           const std::string &htxs_pth,
+                           const std::string &htxs_njets) {
+    auto df1 = df.Define(
+        "ggH_WG1_uncertainties",
```

Comment:
This will cause problems if we run this function within the channel scopes, since the same quantity would be used for the e.g. et and mt. Either we set this string in the python part, or the stxs stuff can only run in the global scope.

Reply:
ah no, i am wrong, we have different frames for each scope right ?

Reply:
right, should be no problem

Reply:
ok, its fine then